core/lib/snapshots_applier/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+            miniblock_number: snapshot.miniblock_number,
+            miniblock_root_hash,
+            storage_logs_chunks_ids_already_processed: vec![],
+            storage_logs_chunks_ids_to_process: chunk_ids,
+        })
+    }
+
+    async fn recover_factory_deps(
+        &mut self,
+        storage: &mut StorageProcessor<'_>,
+    ) -> anyhow::Result<(), SnapshotApplierError> {
+        let factory_deps: SnapshotFactoryDependencies = self
+            .blob_store
+            .get(self.applied_snapshot_status.l1_batch_number)
+            .await?;
+
+        let all_deps_hashmap: HashMap<H256, Vec<u8>> = factory_deps
+            .factory_deps
+            .into_iter()
+            .map(|dep| (hash_bytecode(&dep.bytecode.0), dep.bytecode.0))
+            .collect();
+        storage
+            .storage_dal()
+            .insert_factory_deps(
+                self.applied_snapshot_status.miniblock_number,
+                &all_deps_hashmap,
+            )
+            .await?;
+        Ok(())
+    }
+    async fn insert_initial_writes_chunk(
+        &mut self,
+        storage_logs: &[SnapshotStorageLog],
+        storage: &mut StorageProcessor<'_>,
+    ) -> Result<(), SnapshotApplierError> {
+        tracing::info!("Loading {} storage logs into postgres", storage_logs.len());
+        storage
+            .storage_logs_dedup_dal()
+            .insert_initial_writes_from_snapshot(&storage_logs)
+            .await?;
+        Ok(())
+    }
+    async fn insert_storage_logs_chunk(
+        &mut self,
+        storage_logs: &[SnapshotStorageLog],
+        storage: &mut StorageProcessor<'_>,
+    ) -> Result<(), SnapshotApplierError> {
+        storage
+            .storage_logs_dal()
+            .insert_storage_logs_from_snapshot(
+                self.applied_snapshot_status.miniblock_number,
+                storage_logs,
+            )
+            .await?;
+        Ok(())
+    }
+
+    async fn recover_storage_logs_single_chunk(
+        &mut self,
+        chunk_id: u64,
+    ) -> Result<(), SnapshotApplierError> {
+        let mut storage = self
+            .connection_pool
+            .access_storage_tagged("snapshots_applier")
+            .await?;
+        let mut storage = storage.start_transaction().await?;
+
+        let storage_key = SnapshotStorageLogsStorageKey {
+            chunk_id,
+            l1_batch_number: self.applied_snapshot_status.l1_batch_number,
+        };
+
+        tracing::info!("Processing chunk {chunk_id}");
+
+        let storage_snapshot_chunk: SnapshotStorageLogsChunk =
+            self.blob_store.get(storage_key).await?;
+
+        let storage_logs = &storage_snapshot_chunk.storage_logs;
+        self.insert_storage_logs_chunk(storage_logs, &mut storage)
+            .await?;
+
+        self.insert_initial_writes_chunk(storage_logs, &mut storage)
+            .await?;
+
+        self.applied_snapshot_status
+            .storage_logs_chunks_ids_already_processed
```

Comment:
Don't you need to change `storage_logs_chunks_ids_to_process` as well? See a line comment about the recovery progress representation.